src/context/UserLetterContext.tsx

```diff
@@ -38,29 +53,46 @@ const DefaultUserLetterContext: UserLetterContextType = {
   fromStamp: "",
   content: "",
   type: LetterType.IFrame,
+  currentUser: {
+    name: "",
+    color: randomColor(),
+  },
   setFromName: () => {},
   setToName: () => {},
   setFromStamp: () => {},
   setContent: () => {},
   setType: () => {},
+  setColor: () => {},
   onLetterSubmitted: () => {},
   highestZIndex: 0,
   bumpHighestZIndex: () => {},
+  updateLetterInteraction: () => {},
 };
+
+const DefaultPersistedUserLetterContext: PersistedUserLetterContextInfo = {
+  ...DefaultUserLetterContext,
+  color: DefaultUserLetterContext.currentUser.color,
+};
+
 export const UserLetterContext = createContext<UserLetterContextType>(
   DefaultUserLetterContext
 );
 
 const UserContextStorageId = "user-letter-context";
 
 export function UserLetterContextProvider({ children }: PropsWithChildren) {
+  // const randColor = useMemo(() => randomColor(), []);
   const [userContext, setUserContext] =
     useStickyState<PersistedUserLetterContextInfo>(
       UserContextStorageId,
-      DefaultUserLetterContext
+      DefaultPersistedUserLetterContext
     );
 
-  const { fromName, toName, fromStamp, content, type } = userContext;
+  // TODO: if you want to get ip of user
+  // <script src="https://cdn.jsdelivr.net/gh/joeymalvinni/webrtc-ip/dist/bundle.dev.js"></script>
```

Comment:
what would this be for? uniquely identifying users? maybe browser fingerprinting is better here

Reply:
how do we do browser fingerprinting and yeah something that is not just color for ID

Reply:
https://www.npmjs.com/package/@fingerprintjs/fingerprintjs we could use this?